### ISSUE TITLE
Automatically track a manga

### ISSUE THREAD

**Issue by FeeninForMore** [KotatsuApp/Kotatsu#986]
### Describe your suggested feature

Could there be an option to where instead of having to manually track a manga you could automatically track it?

### Acknowledgements

- [X] This is not a duplicate of an existing issue. Please look through the list of [open issues](https://github.com/KotatsuApp/Kotatsu/issues) before creating a new one.

**Comment by MariusAlbrecht** [comment]
I'm not the maintainer but I think it's unclear what you mean.
Do you want the app to automatically add a new tracking entry when you start reading a new comic? That'd mean it has to determine the mapping between source and tracking which I think is unreasonable.

**Comment by FeeninForMore** [comment]
Yea, well I just suggested a feature I didn't know if it was unreasonable

**Comment by MariusAlbrecht** [comment]
I don't think you're suggesting an actual feature but just a vague idea, that's why I wanted to clarify.
Right now, there sometimes are multiple tracking entries which seem viable to the app for a single source. What would you want the feature to do in that case?
My personal answer is that I'd expect the app to _always_ choose the correct entry, which is just very hard to do (if possible at all).

**Comment by FeeninForMore** [comment]
Well you're not the maintainer so we'll so what he was to say about it

**Comment by Koitharu** [comment]
The main difficulty is to choose the right manga on the tracker. Now you have to select it manually to link manga from the manga source and from the tracker. Ofc it is possible to match it by exact title equality but it would not work in 100% cases.

**Comment by FeeninForMore** [comment]
It's fine then, it's not like I can't do it manually it's just that sometimes I forgot to track my manhwas

**Comment by Neoniq** [comment]
Possible solution is to auto track with first found media entry, if it's wrong then user can manually change it

Sprinkle some toggles for on off on top and done

**Comment by wolfkin** [comment]
So in theory this would add to favorites when you search for a manhwa? or add to favorites when you read a manhwa?

So if I want to for instance auto track Read or Die. 
1. I would open the app. 
2. Search for [read or die]. 
3. There's 5+ different sources that show up. 
4. I pick one and 
5. read a few chapters and then 
6. close the app. 
 
Is the idea that at some point in this process the system would automatically favorite RoD and track it?

Because it seems to me that unless this happens sometime after step 4 for instance the app has no way of knowing which source I want to favorite and sometimes that doesn't matter but sometimes it very much does. Especially with new/current material. Just grabbing the first result would be a sloppy feature even with source ordering/priority. 

In contrast the current history feature is really really useful for things like this. Because in that same scenario where I've done steps 1-6 and forgot to favorite the series. I can always go back to my history and easily find it and favorite it then.

**Comment by MariusAlbrecht** [comment]
The way I understand it, it's only about tracking. The feature will neither choose a source nor will it favourite anything.

You still make an important point I think: _When_ does it automatically track things?
Tracking integration allows for multiple sources to be connected to the same "tracking endpoint" which can cause weird behaviour with the tracking service. Normally that doesn't happen because migration also takes care of that. But if, for example, sources get immediately tracked when any chapter is read that could lead to problems. I often look into multiple sources when starting a new manga.

I see the following solutions here:
1. set a threshold (e.g. only start tracking after 5 read chapters). Could work but won't really be robust
2. do a sort of implicit migration to make sure the auto tracking doesn't connect multiple sources to the same "tracking endpoint". This option would be robust and closer to the behaviour I personally would expect but is also (potentially) harder to implement.

**Comment by MariusAlbrecht** [comment]
The way I understand it, it's only about tracking. The feature will neither choose a source nor will it favourite anything.

You still make an important point I think: _When_ does it automatically track things?
Tracking integration allows for multiple sources to be connected to the same "tracking endpoint" which can cause weird behaviour with the tracking service. Normally that doesn't happen because migration also takes care of that. But if, for example, sources get immediately tracked when any chapter is read that could lead to problems. I often look into multiple sources when starting a new manga.

I see the following solutions here:
1. define some sort of policy on when to start tracking (e.g. only start tracking after 5 read chapters). Could work but won't really be robust
2. do a sort of implicit migration to make sure the auto tracking doesn't connect multiple sources to the same "tracking endpoint". This option would be robust and closer to the behaviour I personally would expect but is also (potentially) harder to implement.
